### PR TITLE
Make two Slide in Slide cron-workers more robust 

### PR DESCRIPTION
**Pull request by danquah**
by moving their risky code inside a try/catch

OSD-105

### CODE REVIEW

File: src/kkos2-display-bundle/Cron/BrugByenSisCron.php [hunk 1]
```diff
@@ -48,16 +46,28 @@ public function getSlideData(SlidesInSlideEvent $event) {
     // Clear errors before run.
     $slide->setOption('cronfetch_error', '');
 
+    try {
+      $this->processSlideEvent($slide);
+    }
+    catch (\Exception $e) {
+      $this->logger->error("Error while processing BrugByenSisCron slide event: " . $e->getMessage());
+      $cronErrors[] = $e->getMessage();
```
Comment (kasperg) [inline src/kkos2-display-bundle/Cron/BrugByenSisCron.php:54]: `$cronErrors[] = $e->getMessage();` seems superfluous. `$cronErrors` is never defined nor used subsequently. Do we want to set/add this value to `$slide->setOption('cronfetch_error', '')`?

File: src/kkos2-display-bundle/Cron/BrugByenSisCron.php [hunk 2]
```diff
@@ -48,16 +46,28 @@ public function getSlideData(SlidesInSlideEvent $event) {
...
+  protected function processSlideEvent(SlidesInSlide $slide) {
     $events = [];
     $cronErrors = [];
     $this->eventfeedHelper->setSlide($slide, 'kk-brugbyen');
-    $data = $this->eventfeedHelper->fetchData();
 
-    $data = $this->eventfeedHelper->sliceData($data);
-    foreach($data as $eventData) {
+    $slices = $this->eventfeedHelper->fetchData();
+    $slice = $this->eventfeedHelper->sliceData($slices);
```
Comment (kasperg) [inline src/kkos2-display-bundle/Cron/BrugByenSisCron.php:64]: Related to my comment above I am not really a fan of multilayered exception handling with the handling being about the same.

Can we keep all the exception handling entirely within the `processSlideEvent`function by also wrapping these two lines in a new try/catch?

**Review by kasperg — APPROVED** [review]
I have a couple of comments.

I would normally consider the `$cronErrors` handling in `getSlideData()` a blocker but I trust you can fix this in a good way yourself 🙂.

File: src/kkos2-display-bundle/Cron/BrugByenSisCron.php (hunk 1 above)
Comment (danquah) [reply src/kkos2-display-bundle/Cron/BrugByenSisCron.php:54]: 🤦  I was a bit to fast here, will fix.

**Comment by danquah** [comment]
aaaand I merged before pushing my fixes ... the fixes will now be merged via #62 instead

**Pull request merged**